app/controllers/ships_controller.rb

```diff
@@ -50,77 +50,86 @@ def info
 
         EjectCargoWorker.perform_async(current_user.id, params[:loader], amount)
-        render(json: {}, status: 200) && (return)
+        render(json: {}, status: :ok) && (return)
       else
-        render(json: { error_message: I18n.t('errors.invalid_amount') }, status: 400) && (return)
+        render(json: { error_message: I18n.t('errors.invalid_amount') }, status: :bad_request) && (return)
       end
     end
-    render json: {}, status: 400
+    render json: {}, status: :bad_request
   end
 
   def insure
     if params[:id]
-      ship = Spaceship.find(params[:id]) rescue nil
-      if ship && (ship.user == current_user) && !ship.insured && current_user.docked
+      ship = Spaceship.ensure(params[:id])
+      if ship && (ship.user == current_user) && !ship.insured? && current_user.docked?
         price = (Spaceship.get_attribute(ship.name, :price) / 2).round
 
-        # check credits
-        render(json: { 'error_message': I18n.t('errors.you_dont_have_enough_credits') }, status: 400) && (return) unless current_user.units >= price
+        # check that they have enough units
+        if price > current_user.units
+          render(json: { 'error_message': I18n.t('errors.you_dont_have_enough_credits') }, status: :bad_request)
+          return
+        end
 
-        # Insure
-        ship.update_columns(insured: true)
+        # Insure & charge them
+        ActiveRecord::Base.transaction do
+          ship.update_columns(insured: true)
+          current_user.reduce_units(price)
+        end
 
-        # Deduct units
-        current_user.reduce_units(price)
-
-        render(json: {}, status: 200) && (return)
+        render(json: {}, status: :ok) && (return)
       end
     end
-    render json: {}, status: 400
+    render json: {}, status: :bad_request
   end
 
   def custom_name
     if params[:name] && params[:id]
-      ship = Spaceship.find(params[:id]) rescue nil
-
-      if ship && (ship.user == current_user) && (params[:name].length <= 15)
-        ship.update_columns(custom_name: params[:name])
-        render(json: {}, status: 200) && (return)
+      ship = Spaceship.ensure(params[:id])
+
+      if ship && (ship.user == current_user)
+        params[:name] = nil if params[:name].blank?
+        if ship.update(custom_name: params[:name])
+          render(json: {}, status: :ok)
+          return
+        end
       end
     end
-    render json: {}, status: 400
+    render json: {}, status: :bad_request
   end
 
   def upgrade_modal
     render partial: 'ships/upgrade_modal', locals: { ship: current_user.active_spaceship }
   end
 
   def upgrade
-    if current_user.docked && current_user.active_spaceship && (current_user.active_spaceship.level < 5)
+    if current_user.docked? && current_user.active_spaceship && (current_user.active_spaceship.level < 5)
```

Comment:
Will the `?` work here? Because it's not an enum

Reply:
Rails boolean fields automatically get the `<field_name>?` method